pco_cli/src/input/audio.rs

```diff
@@ -11,12 +11,21 @@ pub fn get_wav_field(path: &Path) -> Result<Option<Field>> {
   // this is excessively slow, but easy for now
   let mut file = File::open(path)?;
   let (header, _) = wav::read(&mut file)?;
-  let dtype = match header.bytes_per_sample {
-    1 | 2 => Ok(DataType::Int16),
-    3 => Ok(DataType::Int32),
-    4 => Ok(DataType::Float32),
+  let dtype = match (header.audio_format, header.bits_per_sample) {
+    (wav::WAV_FORMAT_PCM, 8 | 16) => Ok(DataType::Int16),
+    (wav::WAV_FORMAT_PCM, 24 | 32) => Ok(DataType::Int32),
+    (wav::WAV_FORMAT_PCM, _) => Err(anyhow!(
+      "unsupported bit depth for WAV_FORMAT_PCM: {}",
+      header.bits_per_sample
+    )),
+    (wav::WAV_FORMAT_IEEE_FLOAT, 32) => Ok(DataType::Float32),
+    (wav::WAV_FORMAT_IEEE_FLOAT, _) => Err(anyhow!(
+      "unsupported bit depth for WAV_FORMAT_IEEE_FLOAT: {}",
+      header.bits_per_sample
+    )),
     _ => Err(anyhow!(
-      "invalid number of bytes per wav file sample"
+      "unsupported audio format: {}",
```

Comment:
Can we just have one failure case instead of 3, formatting in the format (with debug if necessary, `{:?}`).

Reply:
Sure, like this?
``` rust
  let dtype = match (header.audio_format, header.bits_per_sample) {
    (wav::WAV_FORMAT_PCM, 8 | 16) => Ok(DataType::Int16),
    (wav::WAV_FORMAT_PCM, 24 | 32) => Ok(DataType::Int32),
    (wav::WAV_FORMAT_IEEE_FLOAT, 32) => Ok(DataType::Float32),
    _ => Err(anyhow!(
      "unsupported audio format: {:?}",
      header
    )),
  }?;
```
Just thought it'd be more user friendly to provide a more accurate error message, but if you prefer to keep the code simpler that's fine with me :)

Reply:
Ok, technically they're both included above since the entire header is printed.

But we can print only `audio_format` and `bits_per_sample` if you prefer. Is this what you're looking for?
``` rust
  let dtype = match (header.audio_format, header.bits_per_sample) {
    (wav::WAV_FORMAT_PCM, 8 | 16) => Ok(DataType::Int16),
    (wav::WAV_FORMAT_PCM, 24 | 32) => Ok(DataType::Int32),
    (wav::WAV_FORMAT_IEEE_FLOAT, 32) => Ok(DataType::Float32),
    _ => Err(anyhow!(
      "audio format {} with {} bits per sample not supported",
      header.audio_format,
      header.bits_per_sample
    )),
  }?;
```
Otherwise, I think it's easier if you just fix it ;-)